pkg/server/structlogging/structlogging_test: TestHotRangesStats failed

pkg/server/structlogging/structlogging_test.TestHotRangesStats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11869229?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11869229?buildTab=artifacts#/) on master @ [a3dccc6eeb0fb00634806ded691f74f1835694f0](https://github.com/cockroachdb/cockroach/commits/a3dccc6eeb0fb00634806ded691f74f1835694f0):


```
=== RUN   TestHotRangesStats
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/0efd4b21e7a2d6ab8afc437d5552beda/logTestHotRangesStats2293490299
    hot_ranges_log_test.go:86: condition failed to evaluate within 5s: from hot_ranges_log_test.go:99: waiting for logs
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0efd4b21e7a2d6ab8afc437d5552beda/logTestHotRangesStats2293490299
--- FAIL: TestHotRangesStats (15.74s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability @cockroachdb/obs-inf-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHotRangesStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/server/structlogging/structlogging_test.TestHotRangesStats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11869230?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11869230?buildTab=artifacts#/) on master @ [a3dccc6eeb0fb00634806ded691f74f1835694f0](https://github.com/cockroachdb/cockroach/commits/a3dccc6eeb0fb00634806ded691f74f1835694f0):


```
=== RUN   TestHotRangesStats
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/0efd4b21e7a2d6ab8afc437d5552beda/logTestHotRangesStats1723742279
    hot_ranges_log_test.go:86: condition failed to evaluate within 5s: from hot_ranges_log_test.go:99: waiting for logs
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0efd4b21e7a2d6ab8afc437d5552beda/logTestHotRangesStats1723742279
--- FAIL: TestHotRangesStats (16.59s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHotRangesStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/structlogging/hot_ranges_log_test.go
+++ b/pkg/server/structlogging/hot_ranges_log_test.go
@@ -83,7 +83,7 @@ func TestHotRangesStats(t *testing.T) {
 		return nil
 	})
 
-	testutils.SucceedsWithin(t, func() error {
+	testutils.SucceedsSoon(t, func() error {
 		log.FlushFiles()
 		entries, err := log.FetchEntriesFromFiles(
 			0,
@@ -99,5 +99,5 @@ func TestHotRangesStats(t *testing.T) {
 			return errors.New("waiting for logs")
 		}
 		return nil
-	}, 5*time.Second)
+	})
 }

